### ISSUE TITLE
Warn about undeclared path parameters

### ISSUE THREAD

**Issue by LexLuengas** [stoplightio/spectral#1210]
**User story.**
As a developer, I want to detect paths with path parameters that are not declared. 

**Is your feature request related to a problem?**
Without detection, those path parameters remain ambiguous until someone realizes information is missing and complains.

**Describe the solution you'd like**
I would like to see warnings about path parameters that are undeclared.

**Additional context**
For example, given the following paths object
```yaml
/pets/{petId}:
  get:
    summary: Info for a specific pet
    operationId: showPetById
    tags:
      - pets
    responses:
      [...]
```
(i.e. there is no `parameters` property), I would like to see a warning along the lines of:

> warn: the method `get` for path `/pets/{petId}` uses curly braces but contains no parameter declaration.

I am unsure whether this directly relates to #1102.

**Comment by nulltoken** [comment]
@LexLuengas What you describes looks very similar to what the `path-params` rule already performs.

https://github.com/stoplightio/spectral/blob/develop/src/rulesets/oas/functions/__tests__/oasPathParam.test.ts#L32-L59

Is it not working well in your context, or did I misunderstand your request?

**Comment by LexLuengas** [comment]
Thank you. My mistake. I had used version 5.2.0, where the rule didn't exist yet.
